src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/SymbolTable.cs

```diff
@@ -898,11 +898,7 @@ private static List<object> BuildDeclarationChain(Type callingType)
                     ParameterInfo[] parameters = methodBase.GetParameters();
 
                     bool bAdded = false;
-#if UNSUPPORTEDAPI
-                    foreach (MethodInfo methinfo in Enumerable.Where(t.DeclaringType.GetRuntimeMethods(), m => m.MetadataToken == methodBase.MetadataToken))
-#else
                     foreach (MethodInfo methinfo in Enumerable.Where(t.DeclaringType.GetRuntimeMethods(), m => m.HasSameMetadataDefinitionAs(methodBase)))
```

Comment:
I believe its only checking methods that are on the declaring type, which implicitly, is the same as performing a module check.